service/storage_proxy.cc

```diff
@@ -6545,7 +6545,37 @@ db::hints::manager& storage_proxy::hints_manager_for(db::write_type type) {
 }
 
 future<> storage_proxy::truncate_blocking(sstring keyspace, sstring cfname, std::optional<std::chrono::milliseconds> timeout_in_ms) {
-    return remote().send_truncate_blocking(std::move(keyspace), std::move(cfname), timeout_in_ms);
+    locator::effective_replication_map_ptr erm;
+    const auto& ks = _db.local().find_keyspace(keyspace);
+    if (ks.get_replication_strategy().uses_tablets()) {
```

Comment:
I think we don't need this special casing. With vnodes, preventing bootstrap concurrent with truncate is also something we want for safety.

Reply:
I am not sure I understand what you mean. How would we wait for the load balancer to have no tablet migrations for the table we are truncating if the table belongs to a vnode keyspace?

Reply:
If there is no tablet map, the wait is immediately satisfied.

Reply:
I have implemented the suggested change.